### ISSUE TITLE
Selection of stations doesn't take into account dataset d2

### ISSUE THREAD

**Issue by yzchang** [eXascaleInfolab/TSM-Bench#43]
As shown below, the code only selects station IDs in 1..10, probably because the dataset `d1` has 10 stations, but `d2` has 2000 stations.  Since the data is sorted by stations, the generated queries will only address a small portion of data at the beginning of the >172M records.

```
# stations
li = ['st' + str(z) for z in random.sample(range(10), n_st)]
...
temp = temp.replace("<stid>", q)
temp = temp.replace("<stid1>", f"{random.randint(1,10)}")
temp = temp.replace("<stid2>", f"{random.randint(1,10)}")
'''

**Comment by AKheli** [comment]
Thank you for pointing out the issue with station ID selection. We found that the choice of stations had minimal impact on the overall system performance. We have addressed this nonetheless by enabling specifying the number of stations for a given dataset, ensuring that the queries are appropriately scoped. This manual input method provides better control over the data subset used for performance analysis.

**Comment by AKheli** [comment]
Thank you for pointing out the issue with station ID selection. We found that the choice of stations had minimal impact on the overall system performance. We have addressed this nonetheless by enabling specifying the number of stations for a given dataset, ensuring that the queries are appropriately scoped. This manual input method provides better control over the data subset used for performance analysis.
